registration-processor/pre-processor/registration-processor-supervisor-validator-stage/src/test/java/io/mosip/registration/processor/supervisor/SupervisorValidatorTest.java

```diff
@@ -0,0 +1,487 @@
+	}
+
+	@Test(expected = BaseCheckedException.class)
+	public void testinvalidUserInput() throws Exception {
+		ServerError error = new ServerError();
+		error.setMessage("Invalid Date format");
+		List<ServerError> errors = new ArrayList<>();
+		errors.add(error);
+		userResponseDto.setErrors(errors);
+
+		Mockito.when(registrationStatusService.checkUinAvailabilityForRid(any())).thenReturn(true);
+		Mockito.when(restClientService.getApi(any(), any(), anyString(), any(), any())).thenReturn(userResponseDto)
+				.thenReturn(userResponseDto);
+		registrationStatusDto.setRegistrationType("ACTIVATED");
+		supervisorValidator.validate("reg1234", registrationStatusDto, regOsiDto);
+	}
+
+	@Test(expected = ApisResourceAccessException.class)
+	public void tesApisResourceAccessException() throws Exception {
+		ApisResourceAccessException apisResourceAccessException = new ApisResourceAccessException("bad request");
+		Mockito.when(restClientService.getApi(any(), any(), anyString(), any(), any()))
+				.thenThrow(apisResourceAccessException);
+
+		Mockito.when(registrationStatusService.checkUinAvailabilityForRid(any())).thenReturn(true);
+		registrationStatusDto.setRegistrationType("ACTIVATED");
+		supervisorValidator.validate("reg1234", registrationStatusDto, regOsiDto);
+	}
+
+	@Test(expected = ApisResourceAccessException.class)
+	public void testHttpClientErrorException() throws Exception {
+		HttpClientErrorException httpClientErrorException = new HttpClientErrorException(HttpStatus.BAD_REQUEST,
+				"error");
+		ApisResourceAccessException apisResourceAccessException = new ApisResourceAccessException("bad request",
+				httpClientErrorException);
+		Mockito.when(restClientService.getApi(any(), any(), anyString(), any(), any()))
+				.thenThrow(apisResourceAccessException);
+
+		Mockito.when(registrationStatusService.checkUinAvailabilityForRid(any())).thenReturn(true);
+		registrationStatusDto.setRegistrationType("ACTIVATED");
+		supervisorValidator.validate("reg1234", registrationStatusDto, regOsiDto);
+
+	}
+
+	@Test(expected = ApisResourceAccessException.class)
+	public void testHttpServerErrorException() throws Exception {
+		HttpServerErrorException httpServerErrorException = new HttpServerErrorException(HttpStatus.BAD_REQUEST,
+				"error");
+		ApisResourceAccessException apisResourceAccessException = new ApisResourceAccessException("bad request",
+				httpServerErrorException);
+		Mockito.when(restClientService.getApi(any(), any(), anyString(), any(), any()))
+				.thenThrow(apisResourceAccessException);
+
+		Mockito.when(registrationStatusService.checkUinAvailabilityForRid(any())).thenReturn(true);
+		registrationStatusDto.setRegistrationType("ACTIVATED");
+		supervisorValidator.validate("reg1234", registrationStatusDto, regOsiDto);
+
+	}
+
+	/**
+	 * Test supervisor details null.
+	 *
+	 * @throws Exception the exception
+	 */
+	@Test(expected = BaseCheckedException.class)
+	public void testSupervisorDetailsNull() throws Exception {
```

Comment:
Here we should have a test case for UMC in valid also rt ?
Same comment goes for operator also

Reply:
added test case for UMC validation failed in both stages